src/main/java/frc/robot/subsystems/vision/EstimationRunnable.java

```diff
@@ -35,45 +35,43 @@ public EstimationRunnable(String name, VisionCamera camera) {
     this.photonPoseEstimator =
         new PhotonPoseEstimator(
             layout,
-            PoseStrategy.LOWEST_AMBIGUITY,
+            PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
             photonCamera,
             camera.getRobotToCameraTransform());
+    photonPoseEstimator.setMultiTagFallbackStrategy(PoseStrategy.LOWEST_AMBIGUITY);
   }
 
   @Override
   public void run() {
     PhotonPipelineResult photonResults = photonCamera.getLatestResult();
 
-    if (!photonResults.hasTargets()
-        || (photonResults.targets.size() >= 1
-            && photonResults.targets.get(0).getPoseAmbiguity() > APRILTAG_AMBIGUITY_THRESHOLD)) {
-      return;
-    }
+    if (!photonResults.hasTargets()) return;
 
-    if (photonResults.targets.size() >= 1) {
-      PhotonTrackedTarget target = photonResults.getTargets().get(0);
-      Transform3d t3d = target.getBestCameraToTarget();
-      double distance = Math.hypot(t3d.getX(), t3d.getY());
+    photonResults.targets.removeIf(t -> t.getPoseAmbiguity() > APRILTAG_AMBIGUITY_THRESHOLD);
+    photonResults.targets.removeIf(t -> isTargetTooFarAway(t));
 
-      if (distance > 3.3) return;
-    }
+    if (!photonResults.hasTargets()) return;
```

Comment:
Two of the same conditional statements.

Reply:
GitHub went a little crazy here, the only files that are changed for this PR is EstimationRunnable.java and VisionConstants.java.   I'll look at the history again before it gets merged.

For this comment, you are right, we can remove the first check since removeif should work on an empty list ok.

Reply:
Created clean PR with just the vision changes, closing this one.